src/layers/logging.rs

```diff
@@ -86,6 +94,137 @@ impl Layer for LoggingLayer {
     }
 }
 
+/// `LoggingReader` is a wrapper of `BytesReader`, with logging functionality.
+struct LoggingReader {
+    scheme: Scheme,
+    path: String,
+    op: Operation,
+    has_read: u64,
+    inner: BytesReader,
+}
+
+impl LoggingReader {
+    fn new(scheme: Scheme, op: Operation, path: &str, reader: BytesReader) -> Self {
+        Self {
+            scheme,
+            op,
+            path: path.to_string(),
+            has_read: 0,
+            inner: reader,
+        }
+    }
+}
+
+impl AsyncRead for LoggingReader {
+    fn poll_read(
+        mut self: Pin<&mut Self>,
+        cx: &mut Context<'_>,
+        buf: &mut [u8],
+    ) -> Poll<Result<usize>> {
+        match Pin::new(&mut (*self.inner)).poll_read(cx, buf) {
+            Poll::Ready(res) => match res {
+                Ok(n) => {
+                    self.has_read += n as u64;
+                    trace!(target: "opendal::services", "service={} operation={} path={} has_read={} poll read -> got: {}B", self.scheme, self.op, self.path, self.has_read, n);
+                    Poll::Ready(Ok(n))
+                }
+                Err(e) => {
+                    if e.kind() == ErrorKind::Other {
+                        error!(target: "opendal::services", "service={} operation={} path={} has_read={} poll read -> failed: {:?}", self.scheme, self.op, self.path, self.has_read, e);
+                    } else {
+                        warn!(target: "opendal::services", "service={} operation={} path={} has_read={} poll read -> errored: {:?}", self.scheme, self.op, self.path,  self.has_read, e);
+                    }
+                    Poll::Ready(Err(e))
+                }
+            },
+            Poll::Pending => {
+                trace!(target: "opendal::services", "service={} operation={} path={} has_read={} poll -> Pending", self.scheme, self.op, self.path, self.has_read);
```

Comment:
Please keep the same style.

`poll -> Pending` => `poll read -> pending`

Reply:
maybe no "poll read" is better.